src/utils/firestore.ts

```diff
@@ -0,0 +1,25 @@
+import {Firestore} from '@google-cloud/firestore';
+
+let firestoreSingleton: Firestore|null = null;
+
+export function firestore(): Firestore {
+  if (!firestoreSingleton) {
+    throw new Error('Firestore is not initialised.');
+  }
+  return firestoreSingleton;
+}
+
+export function initFirestore(keyFile: string|void) {
```

Comment:
It looks like all calls to `initFirestore()` are almost identical. Can we instead move all the logic to `firestore()`?

If it hasn't been initialised, we initialise the singleton. The only difference I could see was that in tests, we do not want to use env variable credentials. We can easily check which case we are in by checking `process.env.NODE_ENV` to check if its a testing environment.

This will mean that firestore knows how to initialise itself for different environments and users of it simply need to simply call `firestore()`

Reply:
-1 this approach.

I would prefer an API where we force a conscious initialisation and we make it clear when that init is valid and applied vs when it's not.

(i.e. if I call firestore() and then I call firestore('/some-key-file.json') should it use the key file or not?)

Reply:
There's 2 ideas I conflated in there. One was suggesting merging `firestore()` and `initFirestore()` which I understand your objections.

The second is to move all the logic into `initFirestore()`. This would have all the logic in `initFirestore()` (no params), and it'll check for the `test` env, and decide what credentials to load. It wouldn't allow for another key file, but I don't see that being necessary and this will allow the rest of the code to simply call:
```
initFirestore()
firestore()
```